crates/json/src/schema/formats.rs

```diff
@@ -155,6 +157,8 @@ impl Format {
             Self::RelativeJsonPointer => {
                 ValidationResult::from(RELATIVE_JSON_POINTER_RE.is_match(val).unwrap_or(false))
             }
+            Format::Integer => ValidationResult::from(true),
+            Format::Number => ValidationResult::from(true),
```

Comment:
nit: maybe just go with `Self::` to be consistent with the rest of the switch case

Reply:
and do we want to actually validate the values here? perhaps try to parse the values as float / integer and validate them as such

Reply:
My thought here was to leave validation to the individual connector, since different endpoints will have different requirements on what makes a valid integer or number with respect to the number of digits it can have.

Reply:
I think we should do basic validation that it's an integer or number even if we don't bound its length. A regex, perhaps.

Otherwise it's possible to capture data that is truly invalid, and then only find out about it at read-time when you're trying to use it.

Reply:
The `bigdecimal` and `bigint` crates would also be reasonable to add.

We'll need these anyway, in order to extend the semantics of the `sum` and related reduction annotations to cover string-encoded integers / numbers.

Reply:
I actually hadn't been thinking about the "capturing data" side of things at all so yeah that _does_ make sense to have some basic validation. I'll see what I can come up with.